packages/playground/src/utils/migration.ts

```diff
@@ -1,7 +1,57 @@
-import type { BaseModule } from "@threefold/grid_client/dist/es6/modules/base";
-import type { Contract } from "@threefold/tfchain_client";
+import type { Contract, ExtrinsicResult } from "@threefold/tfchain_client";
+
+import { BaseModule } from "../../../grid_client/dist/es6/modules/base";
+
+/**
+ * `_migrateOldFullVMs` is a function is implemented to list all old `FullVM` that listed in the old `Playground` with type `Fullvm`
+ * @param `module` takes the `module` which is an instance of the `BaseModule`.
+ * @returns Applying the extrinsic in the chain with the batch request.
+ */
+async function _migrateOldFullVMs(module: BaseModule) {
+  module.moduleName = "Fullvm"; // Load old deployments that deployed with `Fullvm` type.
+  const deploymentNames = await module._list();
+  const contracts = await Promise.all(
+    deploymentNames.map(deploymentName => module.getDeploymentContracts(deploymentName)),
+  );
+
+  const extrinsics: ExtrinsicResult<Contract>[] = [];
+
+  const loadUpdateExtrinsics = async () => {
+    return await Promise.all(
+      contracts.flat().map(async contract => {
+        const oldData = JSON.parse(contract.deploymentData || "{}") as unknown as {
+          type: string;
+          name: string;
+          projectName: string;
+        };
+
+        if (oldData.type === "vm") {
+          return;
+        }
+
+        oldData.type = "vm";
+
+        const extrinsic = await module.tfClient.contracts.updateNode({
+          id: +contract.contractID,
+          data: JSON.stringify(oldData),
+          hash: contract.deploymentHash,
+        });
+
+        extrinsics.push(extrinsic);
+        BaseModule.newContracts.push(contract);
+      }),
+    );
+  };
```

Comment:
Using await like that not making any use of it's actual power
```suggestion

  const loadUpdateExtrinsics = () => {
    return Promise.all(
      contracts.flat().map(contract => {
        const oldData = JSON.parse(contract.deploymentData || "{}") as unknown as {
          type: string;
          name: string;
          projectName: string;
        };

        if (oldData.type === "vm") {
          return;
        }

        oldData.type = "vm";

        BaseModule.newContracts.push(contract);
        return module.tfClient.contracts.updateNode({
          id: +contract.contractID,
          data: JSON.stringify(oldData),
          hash: contract.deploymentHash,
        });
      }),
    );
  };
```